File: XmlClient/src/app/interfaces/user-professional-details.ts
import { EducationDto } from "./education-dto";
import { ExperienceDto } from "./experience-dto";
import { InterestDto } from "./interest-dto";
import { SkillDto } from "./skill-dto";

export interface UserProfessionalDetails {
    username   : string;
	interests  : InterestDto[];
	skills     : SkillDto[];
	educations : EducationDto[];
	experiences : ExperienceDto[];
}

File: XML/user_service/domain/dto/UserProfessionalDetails.go
package dto

type UserProfessionalDetails struct {
	Username    string          `json:"username"`
	Interests   []InterestDto   `json:"interests"`
	Skills      []SkillDto      `json:"skills"`
	Educations  []EducationDto  `json:"educations"`
	Experiences []ExperienceDto `json:"experiences"`
}
